homework/index.js

```diff
@@ -30,17 +32,94 @@
+    for (let i = 0; i < x.length; i++) {
+      createAndAppend('option', select, { text: x[i].name, value: i });
+    }
+  };
+
+  const div = createAndAppend('div', root, { id: 'container' });
+  const leftContainer = createAndAppend('div', div, { class: 'left-block frame' });
+  const table = createAndAppend('table', leftContainer);
+  const listContributor = createAndAppend('tbody', table);
+
+  const rightContainer = createAndAppend('div', div, { class: 'right-block frame' });
+  createAndAppend('p', rightContainer, {
+    text: 'Contributions',
+    class: 'contributor',
+  });
+
+  const list = createAndAppend('ul', rightContainer, { class: 'list1' });
+
+  function renderContributors(selRep) {
+    list.innerHTML = '';
+
+    for (let i = 0; i < selRep.length; i++) {
+      const li = createAndAppend('li', list, { class: 'contributor-item' });
+      const linkFor = createAndAppend('a', li, { target: '_blank', href: selRep[i].html_url });
+      createAndAppend('img', linkFor, { src: selRep[i].avatar_url, class: 'avatar', height: 48 });
+      const divCont = createAndAppend('div', linkFor, { class: 'contributor-data' });
+      createAndAppend('div', divCont, { text: selRep[i].login });
+      createAndAppend('div', divCont, {
+        text: selRep[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  function renderRepo(listContributors, repo) {
+    const content = [
+      { title: 'Repository', attribute: 'name' },
+      { title: 'Description', attribute: 'description' },
+      { title: 'Forks', attribute: 'forks' },
+      { title: 'Updated', attribute: 'updated_at' },
+    ];
+    listContributors.innerHTML = '';
+
+    for (let i = 0; i < content.length; i++) {
+      const headTitle = createAndAppend('tr', listContributors);
+      createAndAppend('td', headTitle, { text: `${content[i].title} :`, class: 'label' });
+      const cellContent = createAndAppend('td', headTitle);
+      if (content[i].attribute === 'name') {
+        createAndAppend('a', cellContent, {
+          href: repo.html_url,
+          text: repo.name,
+          target: '_blank',
+        });
+      } else {
+        cellContent.textContent = repo[content[i].attribute];
+      }
+    }
+    fetchJSON(repo.contributors_url, (error, contributors) => renderContributors(contributors));
+  }
+
   function main(url) {
     fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
+      const root1 = document.getElementById('root');
       if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        createAndAppend('div', root1, { text: err.message, class: 'alert-error' });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        let optionArr = data;
+        optionArr = optionArr.sort((a, b) => a.name.localeCompare(b.name));
+
+        option(optionArr);
+
+        const e = document.getElementById('selector');
```

Comment:
Name of the variable isn't the best here. It's hard to understand what `e` represents. 🙂

Reply:
I forget to change it, when I was trying it for it to be quick I just named it as e, let me change it.

Reply:
Maybe you can put it as a task for your homework separately. When you think you are done with your future homeworks, just go through the code you have written and see if there is any room for improvement. I use this approach regularly at work. 🙂